src/main/java/alibek/User.java

```diff
@@ -0,0 +1,33 @@
+package alibek;
+
+import java.util.*;
+
+
+public class User {
+    String name;
+    User(String name) throws Exception {
+        validateName(name);
+        this.setName(name);
+    }
+    User() {
+
+    }
+    public String getName() {
+        return this.name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+    public boolean isMe(String name) {
+        return this.name.equals(name);
+    }
+    public void validateName(String name) throws Exception {
```

Comment:
&#x1F34E; **Issue**
Validation logic is better placed somewhere outside the User class. You validate "name" first, then if it is valid you create User object.